components/identity-mgt/org.wso2.carbon.identity.mgt.endpoint.util/src/main/java/org/wso2/carbon/identity/mgt/endpoint/util/client/SelfRegistrationMgtClient.java

```diff
@@ -210,60 +211,86 @@ public Integer checkUsernameValidity(String username) throws SelfRegistrationMgt
     public Integer checkUsernameValidity(String username, boolean skipSignUpCheck) throws
             SelfRegistrationMgtClientException {
 
+        User user = new User();
+        user.setUsername(username);
```

Comment:
don't we need to set tenantDomain to the user object?

Reply:
This was the method used for preprocessed usernames. We can deprecate this method.

Reply:
let's deprecate.. add the deprecate annotation and add a comment to use new method